test/application_test.rb

```diff
@@ -6,8 +6,8 @@
   end
 
   it 'instantiate new configuration object when inherited' do
-    backend_app  = Backend::Application.new
-    frontend_app = Frontend::Application.new
+    backend_app  = Back::Application.new
```

Comment:
IMHO, keeping the naming then namespaced the constant, for example: `MyApp::Backend::Application`